tests/utils/test_request_utils.py

```diff
@@ -58,3 +59,33 @@ def test_download_chunk_incomplete_read(tmp_path):
                 download_path=download_path,
                 http_uri="https://example.com",
             )
+
+
+@pytest.mark.parametrize("env_value", ["1", "true", "True", "TRUE"])
+def test_redirects_disabled_if_env_var_set(env_value):
+    from requests.exceptions import HTTPError
+
+    os.environ["MLFLOW_DISABLE_HTTP_REDIRECTS"] = env_value
+
+    with mock.patch("requests.Session.request") as mock_request:
+        mock_request.return_value.status_code = 302
+        mock_request.return_value.text = "mock response"
+
+        with pytest.raises(HTTPError, match="HTTP redirects are disabled"):
+            request_utils.cloud_storage_http_request("GET", "http://localhost:5000")
```

Comment:
/ai rewrite this using pytest's `monkeypatch` fixture and explain why that's preferred.